tianshou/utils/net/common.py

```diff
@@ -9,35 +10,74 @@ class Net(nn.Module):
     """Simple MLP backbone. For advanced usage (how to customize the network),
     please refer to :ref:`build_the_network`.
 
-    :param concat: whether the input shape is concatenated by state_shape
+    :param bool concat: whether the input shape is concatenated by state_shape
         and action_shape. If it is True, ``action_shape`` is not the output
         shape, but affects the input shape.
+    :param bool dueling: whether to use dueling network to calculate Q values
+        (for Dueling DQN), defaults to False.
+    :param bool layer_norm: whether to use layer norm before ReLU, defaults to
+        False.
     """
 
-    def __init__(self, layer_num, state_shape, action_shape=0, device='cpu',
-                 softmax=False, concat=False, hidden_layer_size=128):
+    def __init__(self, layer_num: int, state_shape: tuple,
+                 action_shape: Optional[tuple] = 0,
+                 device: Union[str, torch.device] = 'cpu',
+                 softmax: bool = False, concat: bool = False,
+                 hidden_layer_size: int = 128,
+                 dueling: Tuple[int, int] = None, layer_norm: bool = False):
         super().__init__()
         self.device = device
+        self.dueling = dueling
+        self.softmax = softmax
         input_size = np.prod(state_shape)
         if concat:
             input_size += np.prod(action_shape)
-        self.model = [
-            nn.Linear(input_size, hidden_layer_size),
-            nn.ReLU(inplace=True)]
+
+        def miniblock(inp, oup):
```

Comment:
I would have duplicated thus block instead of wrapping it in a small method specifically here, because to me it is more readable but as you want.

Reply:
I don't think so. You can have a look at the left side: https://github.com/thu-ml/tianshou/pull/170/commits/0d1f0a4a3c8b79311fc2e32f90b52b59c034668a

Reply:
Yes I see, and I had no problem with it.